src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_UnmanagedCallersOnly.vb

```diff
@@ -0,0 +1,404 @@
+        Inherits BasicTestBase
+
+        Private ReadOnly _parseOptions As CSharp.CSharpParseOptions = CSharp.CSharpParseOptions.Default.WithLanguageVersion(CSharp.LanguageVersion.Default)
+        Private ReadOnly _csharpCompOptions As CSharp.CSharpCompilationOptions = New CSharp.CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, allowUnsafe:=True)
+        Private ReadOnly _csharpReferences As ImmutableArray(Of MetadataReference) = TargetFrameworkUtil.NetCoreAppReferences.Add(NetCoreApp.SystemRuntimeInteropServices)
+
+        Private ReadOnly UnmanagedCallersOnlyAttributeIl As String = <![CDATA[
+.class public auto ansi sealed beforefieldinit System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute
+    extends [mscorlib]System.Attribute
+{
+    .custom instance void [mscorlib]System.AttributeUsageAttribute::.ctor(valuetype [mscorlib]System.AttributeTargets) = (
+        01 00 40 00 00 00 01 00 54 02 09 49 6e 68 65 72
+        69 74 65 64 00
+    )
+    .field public class [mscorlib]System.Type[] CallConvs
+    .field public string EntryPoint
+
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        ldarg.0
+        call instance void [mscorlib]System.Attribute::.ctor()
+        ret
+    }
+}
+]]>.Value
+
+        <Fact>
+        Public Sub UnmanagedCallersOnlyInSource()
+            Dim source =
+<compilation>
+    <file><![CDATA[
+Imports System.Runtime.CompilerServices
+Imports System.Runtime.InteropServices
+Class C
+    <UnmanagedCallersOnly>
+    Sub S1()
+        S1()
+    End Sub
+
+    <UnmanagedCallersOnly(CallConvs := { GetType(CallConvCdecl) })>
+    Sub S2()
+        S2()
+    End Sub
+
+    Property P1 As String
+        <UnmanagedCallersOnly>
+        Get
+            Return ""
+        End Get
+        <UnmanagedCallersOnly>
+        Set
+        End Set
+    End Property
+
+    Sub S3()
+        P1 = P1
+    End Sub
+End Class
+]]>
+    </file>
+</compilation>
+
+            Dim comp = CreateCompilation(source, references:={NetCoreApp.SystemRuntimeInteropServices}, targetFramework:=TargetFramework.NetCoreApp)
+
+            comp.AssertTheseDiagnostics(<![CDATA[
+BC42365: 'UnmanagedCallersOnly' is not supported.
+    <UnmanagedCallersOnly>
+     ~~~~~~~~~~~~~~~~~~~~
+BC42365: 'UnmanagedCallersOnly' is not supported.
+    <UnmanagedCallersOnly(CallConvs := { GetType(CallConvCdecl) })>
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+BC42365: 'UnmanagedCallersOnly' is not supported.
+        <UnmanagedCallersOnly>
+         ~~~~~~~~~~~~~~~~~~~~
+BC42365: 'UnmanagedCallersOnly' is not supported.
+        <UnmanagedCallersOnly>
+         ~~~~~~~~~~~~~~~~~~~~
+]]>)
+        End Sub
+
+        <Fact>
+        Public Sub UnmanagedCallersOnlyInCsharp()
+            Dim source0 = <![CDATA[
+using System.Runtime.InteropServices;
+public class C
+{
+    [UnmanagedCallersOnly]
+    public static void M1() { }
```

Comment:
>public static void M1() { } [](start = 4, length = 27)

Consider explicitly asserting that this symbol has a use site error and also has unsupported metadata.

Reply:
Did this for every from metadata test as well.

---
In reply to: [566929962](https://github.com/dotnet/roslyn/pull/50886#discussion_r566929962) [](ancestors = 566929962)